src/shared/bpf-link.h

```diff
@@ -3,10 +3,14 @@
 #pragma once
 
 #include <bpf/libbpf.h>
+#include <stddef.h>
```

Comment:
> Ack, I mixed the two. I think this gotcha could be a source of errors in the future, but let's leave it like this of rnow.

Let's mark custom helpers from `src/shared/bpf-program.h` with `_sd_deprecated_` after this lands?

Reply:
> > Ack, I mixed the two. I think this gotcha could be a source of errors in the future, but let's leave it like this of rnow.
> 
> Let's mark custom helpers from `src/shared/bpf-program.h` with `_sd_deprecated_` after this lands?

I'd actually go in the opposite direction: never use the errno-returning apis from libbpf in our code, but make our internal functions thin wrappers around the libbpf code. We do that for various glibc apis (`*_safe()`) already and this helps to reduce the number of bugs.

Reply:
@keszybz 

Thanks for reviewing! Happy to get it merged.

> I'd actually go in the opposite direction: never use the errno-returning apis from libbpf in our code, but make our internal functions thin wrappers around the libbpf code. We do that for various glibc apis (*_safe()) already and this helps to reduce the number of bugs.

This will make libbpf a dependency for the features already implemented and non-requiring libbpf since they use raw BPF instructions, namely `DeviceAllow`, `IPAddress{Allow|Deny}`.
Though libbpf is already [well-represented](https://github.com/libbpf/libbpf#distributions) in distros, systemd customers may get angry by unexpected dynamic dependency.
If that is fine, I agree with thin wrappers approach.

Reply:
I think that would be fine - and we can make it a dlopen dependency, so that minimal build/container folks are not too upset